packages/harmony/src/components/input/TextInput/TextInput.module.css

```diff
@@ -0,0 +1,147 @@
+.root {
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+}
+
+.inputRoot {
+  display: flex;
+  align-items: center;
+  gap: var(--harmony-unit-3);
+  padding: var(--harmony-unit-4);
+  border: 1px solid var(--harmony-border-default);
+  border-radius: var(--harmony-unit-1);
+  background-color: var(--harmony-bg-surface-1);
+  transition: border ease-in-out 0.1s;
+  box-sizing: border-box;
+  & svg path {
+    fill: var(--harmony-text-subdued);
+  }
+}
+
+.small {
+  --gap: -4px;
+  --font-size: var(--harmony-font-s);
+  height: 34px;
+}
+
+.default {
+  --gap: 0px;
+  --font-size: var(--harmony-font-m);
+  height: 64px;
+}
+
+.large {
+  --gap: var(--harmony-unit-1);
+  --font-size: var(--harmony-font-xl);
+  height: 66px;
+}
+
+.root:hover {
+  border-color: var(--harmony-neutral-light-6);
+}
+
+.focused {
+  border-color: var(--harmony-secondary);
+}
+
+.warning,
+.warning.focused {
+  border-color: var(--harmony-orange);
+}
+
+.error,
+.error.focused {
+  border-color: var(--harmony-red);
+}
+
+.error:hover,
+.error:hover.focused {
+  /* TODO: Figure out of we want a hover color */
+  border-color: var(--harmony-red);
+}
+
+.disabled {
+  background-color: var(--harmony-neutral-light-9);
+  border-color: var(--harmony-neutral-light-8);
+}
+
+.textInput {
+  width: 100%;
+  height: 100%;
+  outline: 0;
+  border: 0;
+  background: none;
+  font-size: var(--harmony-font-size);
+  font-weight: var(--harmony-font-medium);
+  color: var(--harmony-text-default);
+}
+
+.textInput::placeholder,
+.label {
+  color: var(--harmony-neutral-light-4);
+  font-weight: var(--harmony-font-medium);
+}
+
+.inputRow {
+  width: 100%;
+  justify-content: space-between;
+}
+
+.inputContainer {
+  flex: 1 1 0;
+}
+
+/** 
+ * Flex container so that the absolutely positioned elevated placeholder
+ * starts out centered vertically
+ **/
+.elevatedLabel {
+  position: relative;
+  display: flex;
+  align-items: center;
+  width: 100%;
+  height: 100%;
+  font-size: var(--harmony-font-size);
+}
+
+/** Add the "*" to required fields, but have it disappear when elevated **/
+.required:not(.focused) .elevatedLabel .placeholder:not(.hasValue)::after {
+  content: ' *';
+}
+
+/** Push the input down a bit to make room for the elevated placeholder **/
+.elevatedLabel .inputRow {
+  padding-top: calc(var(--harmony-font-xs));
+}
+
+/** Position the elevated label absoutely on top of the input **/
+.label {
+  color: var(--harmony-text-subdued);
+  position: absolute;
+  z-index: 2;
```

Comment:
We may want to be careful about z-indexes. Is there risk of this being hidden under or showing through other elements in the app if they have z-index specified?

Reply:
Great callout. I actually ran into this when we made our text input at Bestow. 
I added `isolation: isolate` to the root so that the z-index doesn't spill over anywhere
https://developer.mozilla.org/en-US/docs/Web/CSS/isolation